File: python/src/SelectionSort.py
def findSmallest(array, startingIndex):
    smallestIndex = startingIndex
    for i in range(startingIndex, len(array)):
        if array[i] < array[smallestIndex]:
            smallestIndex = i
    return smallestIndex


array = [22, 45, 2, 78, 56, 99, 1]
for i in range(0, len(array)-1):
    smallestIndex = findSmallest(array, i+1)
    buf = array[i]
    array[i] = array[smallestIndex]
    array[smallestIndex] = buf

for elem in array:
    print(elem)

File: java/src/SelectionSort.java
public class SelectionSort {
    public static void main(String[] args) {
        var array = new int[]{22, 1, 3, 99, 8};
        for (int i = 0; i < (array.length - 1); i++) {
            var smallestIndex = findSmallest(array, i);
            var buf = array[i];
            array[i] = array[smallestIndex];
            array[smallestIndex] = buf;
        }
        for (var elem : array) {
            System.out.println(elem);
        }
    }

    public static int findSmallest(int[] array, int startingIndex) {
        var smallest = startingIndex;
        for (int i = startingIndex; i < array.length; i++) {
            if (array[i] < array[smallest]) {
                smallest = i;
            }
        }
        return smallest;
    }
}
